src/cephadm/cephadm

```diff
@@ -2832,6 +3087,28 @@ def _get_parser():
         'check-host', help='check host configuration')
     parser_check_host.set_defaults(func=command_check_host)
 
+    parser_add_repo = subparsers.add_parser(
+        'add-repo', help='configure package repository')
+    parser_add_repo.set_defaults(func=command_add_repo)
+    parser_add_repo.add_argument(
+        '--release',
+        help='use specified upstream release')
+    parser_add_repo.add_argument(
+        '--dev',
+        help='use specified bleeding edge build from git branch or tag')
+    parser_add_repo.add_argument(
+        '--dev-commit',
+        help='use specified bleeding edge build from git commit')
```

Comment:
Hmm.  The old cpeh-deploy install function supports:
```
optional arguments:
  -h, --help            show this help message and exit
  --stable [CODENAME]   [DEPRECATED] install a release known as CODENAME (done
                        by default) (default: None)
  --release [CODENAME]  install a release known as CODENAME (done by default)
                        (default: None)
  --testing             install the latest development release
  --dev [BRANCH_OR_TAG]
                        install a bleeding edge build from Git branch or tag
                        (default: master)
  --dev-commit [COMMIT]
                        install a bleeding edge build from Git commit
                        (defaults to master branch)
  --mon                 install the mon component only
  --mgr                 install the mgr component only
  --mds                 install the mds component only
  --rgw                 install the rgw component only
  --osd                 install the osd component only
  --tests               install the testing components
  --cli, --common       install the common component only
  --all                 install all Ceph components (mon, osd, mds, rgw)
                        except tests. This is the default
  --adjust-repos        install packages modifying source repos
  --no-adjust-repos     install packages without modifying source repos
  --repo                install repo files only (skips package installation)
  --local-mirror [LOCAL_MIRROR]
                        Fetch packages and push them to hosts for a local repo
                        mirror
  --repo-url [REPO_URL]
                        specify a repo URL that mirrors/contains Ceph packages
  --gpg-url [GPG_URL]   specify a GPG key URL to be used with custom repos
                        (defaults to ceph.com)
  --nogpgcheck          install packages without gpgcheck
```
Since this function only controls the repo itself, most of the install-related options go away.  But --*-url and --nopgpcheck make sense...